### DISCUSSION TITLE
Security step for changing of the password to a KeePassXC database

### DISCUSSION THREAD

**Discussion by KodeXall** [keepassxreboot/keepassxc#9320] [Ideas]
When changing password to a Database, why is not there a request for providing of the old password? Like to virtually every other service or on mos of websites?

If somebody takes control over my machine with my database being opened in KeePassXC, such person then can lock me out for good from my own database file

Yes- I do have multiple copies. But I can imagine also a situation when wrongdoer robs me also of all of my backups and / or my copies are so old that I do not remember / store password for them anymore

**Comment by droidmonkey** [comment]
> If somebody takes control over my machine with my database being opened in KeePassXC

You have lost. Full stop. There is nothing we can do to prevent your loss of sensitive information.

**Comment by phoerious** [comment]
KeePassXC is not a website or "service". It's an encrypted database. There is no barrier preventing that an attacker can change your password. An additional prompt for the old password is mere security theatre without any real protection.
